src/naive/datetime/mod.rs

```diff
@@ -131,6 +131,101 @@ impl NaiveDateTime {
         datetime.expect("invalid or out-of-range datetime")
     }
 
+    /// Makes a new `NaiveDateTime` corresponding to a UTC date and time,
+    /// from the number of non-leap milliseconds
+    /// since the midnight UTC on January 1, 1970 (aka "UNIX timestamp")
+    ///
+    /// Returns `None` on the out-of-range number of milliseconds.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use chrono::{NaiveDateTime, NaiveDate};
+    ///
+    ///
+    /// let dt = NaiveDate::from_ymd(1970, 1, 1).and_hms_milli(0, 0, 0, 42);
+    ///
+    /// let got = NaiveDateTime::from_timestamp_millis(dt.timestamp_millis()).unwrap();
+    /// assert_eq!(dt, got);
+    /// let dt = NaiveDate::from_ymd(3015, 9, 5).and_hms(23, 56, 4);
+    /// let ms = dt.timestamp_millis();
+    /// assert_eq!(NaiveDateTime::from_timestamp_millis(ms), Some(dt));
+    /// ```
+    #[inline]
+    pub fn from_timestamp_millis<T: Into<i64>>(milliseconds: T) -> Option<NaiveDateTime> {
```

Comment:
I'd rather stay away from the `Into` argument type here, instead using `i64` directly. Same for the other new methods.